File: module-domain/src/main/java/eu/geekhome/domain/configurable/IntegerFieldBuilder.java
package eu.geekhome.domain.configurable;

public class IntegerFieldBuilder implements FieldBuilder<Integer> {
    @Override
    public Integer fromPersistableString(String value) {
        return Integer.parseInt(value);
    }

    @Override
    public String toPersistableString(Integer value) {
        return value.toString();
    }
}

File: module-domain/src/main/java/eu/geekhome/domain/configurable/IntegerFieldBuilder.kt
package eu.geekhome.domain.configurable

class IntegerFieldBuilder : FieldBuilder<Int> {
    override fun fromPersistableString(value: String?): Int {
        if (value == null) {
            return 0
        }

        return value.toInt()
    }

    override fun toPersistableString(value: Int): String {
        return value.toString()
    }
}